lib/tasks/asset_sync.rake

```diff
@@ -20,11 +20,24 @@ if Rake::Task.task_defined?("assets:precompile:nondigest")
     Rake::Task["assets:sync"].invoke if defined?(AssetSync) && AssetSync.config.run_on_precompile
   end
 else
+  # Triggers on webpacker compile task instead of assets:precompile to ensure
```

Comment:
> The intention is to ensure that we enhance either the "webpacker:compile" or "assets:precompile" tasks but never both.

Agreed, and I think both the implementation that "inspired" this PR does that with FashionThing's setup (since `Rake::Task.task_defined?("webpacker:compile")` and `AssetSync.config.include_webpacker_assets` will both be true), so I'll leave it as is. 👍